classes/class3/Homework/HomeworkApp/app.js

```diff
@@ -0,0 +1,129 @@
+//initializing packages
+var express = require('express');
+var path = require('path');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var exphbs = require ('express-handlebars');
+
+//requiring modules
+var index = require('./routes/index');
+var app = express();
+
+//selecting port
+var PORT = process.env.PORT || 3000;
+
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+
+
+app.use(logger('dev'));  //call logger function and pass it 'dev'
+app.use(bodyParser.json());  //parses body of request and puts into json object
+app.use(bodyParser.urlencoded({ extended: false }));  //parses url things
+app.use(cookieParser()); //parses cookies
+app.use(express.static(path.join(__dirname, 'public')));  //static page
+
+
+//initializing mongoose
+var mongoose = require('mongoose');
+var mongoURI = process.env.MONGOURI || "mongodb://localhost/test";
+//var mongoURI = 'mongodb://user:password@ds053638.mongolab.com:53638/olinjs' || "mongodb://localhost/test";
+mongoose.connect(mongoURI);
+var db = mongoose.connection
+//db.on('error', console.error.bind(console, 'connection error:'));
+db.once('open',function (callback) {
+	var catSchema = mongoose.Schema( {
+		name: String, 
+		age: Number, 
+		colors: Array
+	});
+
+	var Cat = mongoose.model('Cat', catSchema);
+
+	//routing
+
+	app.get('/', index.home);
+
+	app.get('/cats', function(req, res) {
+		//TODO- sort by age and put in pretty handlebars
+		Cat.find().sort("age").exec(function (err, cats) {
+			if (err) {
+				return console.error("There are no cats")
```

Comment:
The database error would not be a result of running out of cats, more likely a connection issue.